### PR TITLE
adds server setup and tests

### PR DESCRIPTION
**Pull request by swells2020**

### CODE REVIEW

File: __tests__/server.test.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+const request = require('supertest')
+const app = require('../server');
+const db = require('../db/connection');
+const data = require('../db/data/test-data/index');
+const seed = require('../db/seeds/seed');
+
+beforeAll(() => {
+    return seed(data).then();
```
Comment (Poonam-raj) [inline __tests__/server.test.js]: .then() not needed here

Comment (Poonam-raj) [reply __tests__/server.test.js]: Also beforeEach we should be seeding rather than beforeAll

File: __tests__/server.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+});
+
+describe('app', () => {
+    test('tests the server connection', () => {
+        return request(app)
+            .get('/api')
+            .expect(200)
+            .then();
+    })
+    test('tests the database connection', () => {
```
Comment (Poonam-raj) [inline __tests__/server.test.js]: You have two describe blocks which seem to be checking the same thing, why do the first if you're essentially re-checking it in the second. I'd get rid of the first if that's the case.

File: __tests__/server.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                expect(testArray.includes('topics')).toBe(true);
+            })
+    })
+    test('tests bad path error handler', () => {
+        return request(app)
+            .get('/invalid-path')
+            .expect(404)
+            .then(response => {
+                expect(response.body).toEqual({ message: '404: invalid path' })
+            })
```
Comment (Poonam-raj) [inline __tests__/server.test.js:38]: Very nice inclusion of bad path testing

**Review by Poonam-raj — APPROVED** [review]
Have a look at my comments around your testing but otherwise all good.I'd make the relevant changes and merge once done

File: __tests__/server.test.js (hunk 1 above)
Comment (swells2020) [reply __tests__/server.test.js]: Updated both.

File: __tests__/server.test.js (hunk 2 above)
Comment (swells2020) [reply __tests__/server.test.js]: I've removed the first test as suggested and tweaked the description in the second test.

File: __tests__/server.test.js (hunk 3 above)
Comment (swells2020) [reply __tests__/server.test.js:38]: Thanks =).

**Pull request merged**